release candidate playtester pinging

corinburger opened the pull request:
could probably look better

![image](https://github.com/itscynxx/Spectre/assets/77130743/725a7bf6-dddd-467d-9c85-b2c906b000ab)
![image](https://github.com/itscynxx/Spectre/assets/77130743/5d59fb6e-07bf-43fb-a343-0a5c86e42ee2)

Commit:
release candidate playtester pinging

itscynxx commented:
this does actually look much better than anything I've done in testing

@GeckoEidechse as this was your requested feature

corinburger committed:
makes it wait until build finishes

waits until northstarlauncher release to send message, fyi i did not test this so

Commit:
changed playtester role id, remove useless print

GeckoEidechse commented:
Holy fuck the embed changelog stuff looks cool :D

GeckoEidechse commented:
I would say, have like a delay before posting the message.

When I push the tag, there's two main things that take some time to complete:
- GitHub Action needs to build release (~10 minutes in worst case)
- Thunderstore cache needs to flush (~20 minutes in worst case)

So if we ping right after tag push, in the first half an hour, playtesters will still receive the old build.

That's why I'd suggest having like a wait time inbetween. Not sure if current libraries in Spectre have like job schedule thing or alternatively, simply do like a `thread.sleep(1800)` lol

Commit:
waits until thunderstore release

ashleyburgers commented:
now it will look for NorthstarReleaseCandidate releases in #mod-releases-thunderstore and then send the message.

also it gets the version name from the newest NorthstarLauncher release on github

heres how the message looks again. i edited the text at the top a bit. the part at the bottom shouldn't show if the latest discussion is draft patch notes
![image](https://github.com/itscynxx/Spectre/assets/77130743/ad7abf63-45d0-4edc-99d7-64845748ea44)

ashleyburgers commented:
@GeckoEidechse not sure how github notifications work so if you already got one for my last comment my bad

GeckoEidechse commented:
Me likey :eyes: 

@itscynxx we should test drive this on the Spectre testing server for one release-candidate or so and if it doesn't go up in flames, then deploy on Northstar Discord :>

itscynxx commented:
@GeckoEidechse nice timing for a ping

can definitely test later, I can test any time between like 5 hours and 10 hours from now

itscynxx commented:
I forgot to set this up but there probably won't be a playtest for a little bit anyways 

will set it up for testing in the testing server tonight

GeckoEidechse commented on cogs/PlaytesterPing.py:
```diff
@@ -0,0 +1,98 @@
+        if message.author == self.bot.user:
+            return
+
+        if message.channel.id == 939573786355859498: #mod-releases-thunderstore
```
Not sure how the rest of Spectre code looks but it would probably be best to define channel ids as a const/global at the top of the file so that it's easier to move them to a config file in the long term ^^

ashleyburgers replied:
other files dont seem to do that, i just made the thunderstore release channel a variable and moved it along with the playtester ping channel variable up a bit

GeckoEidechse commented on cogs/PlaytesterPing.py:
```diff
@@ -0,0 +1,98 @@
+            if re.search(r'NorthstarReleaseCandidate v\d+.\d+.\d+', message.embeds[0].title):
+                data = getLatestDiscussion()
+                rcVersion = getLatestReleaseName()
+                playtestPingChannel = self.bot.get_channel(936678773150081055)
```
Ditto ^^

GeckoEidechse commented on cogs/PlaytesterPing.py:
```diff
@@ -0,0 +1,98 @@
+
+                embed.set_author(name="Northstar " + rcVersion, icon_url="https://avatars.githubusercontent.com/u/86304187")
+                
+                pingMessage = await playtestPingChannel.send(f"<@&936669179359141908>s, there is a new Northstar release candidate, `{rcVersion}`. If you find any issues or have feedback, please inform us in the thread attached to this message.\n\n**Installation**:\nGo to settings in FlightCore, and enable testing release channels. After you've done that, go to the play tab, click the arrow next  ...
```
It might make sense to phrase it such that it's clear that one needs to opt into release candidates only once ^^

ashleyburgers replied:
added that in parentheses 
"Go to settings in FlightCore, and enable testing release channels. (you only need to do this once)"

GeckoEidechse commented:
closes #33

itscynxx commented on cogs/PlaytesterPing.py:
```diff
@@ -0,0 +1,98 @@
+    try:
+        response = requests.get("https://api.github.com/repos/R2Northstar/NorthstarLauncher/releases", headers=headers)
+        if response.status_code == 200:
+            reponse = response
```
What exactly is the reasoning for making `response` read as `reponse`? (don't know the actual wording)

Especially because it doesn't seem to get used

It does work regardless, I'm just curious of the reasoning

ashleyburgers replied:
no idea to be honest, just removed the whole `if response.status_code == 200` part.

itscynxx reviewed (commented):
Works in testing, looks good, works better than anything currently written, just have a question

Commit:
changes

Commit:
change installation text slightly

Commit:
change ids back to northstar server ones

ashleyburgers commented:
don't know why it wouldn't, but i tested it again after changes and its still working

itscynxx commented on cogs/PlaytesterPing.py:
```diff
@@ -0,0 +1,98 @@
+
+                embed.set_author(name="Northstar " + rcVersion, icon_url="https://avatars.githubusercontent.com/u/86304187")
+                
+                pingMessage = await playtestPingChannel.send(f"<@&936669179359141908>s, there is a new Northstar release candidate, `{rcVersion}`. If you find any issues or have feedback, please inform us in the thread attached to this message.\n\n**Installation**:\nGo to settings in FlightCore, and enable testing release channels. (you only need to do this once) After you've done that, go to the ...
```
```suggestion
                pingMessage = await playtestPingChannel.send(f"<@&936669179359141908>, there is a new Northstar release candidate, `{rcVersion}`. If you find any issues or have feedback, please inform us in the thread attached to this message.\n\n**Installation**:\nGo to settings in FlightCore, and enable testing release channels (you only need to do this once). After you've done that, go to the play tab, click the arrow next to `LAUNCH GAME`, and select `Northstar release candidate`. Then, click the `UPDATE` button.", embed=embed)
```

itscynxx reviewed (commented):
Minor grammar nitpicks

I do see the appeal of using "playtesters" over "playtester", but I think it just looks a little odd with how Discord pings work

After this, I'll probably merge it, then wait for Gecko's green light to enable it fully in the server. Thanks for taking the time to fix something I couldn't :D

Commit:
Update PlaytesterPing.py

Co-authored-by: Cyn <70904206+itscynxx@users.noreply.github.com>

itscynxx commented:
@GeckoEidechse any other suggestions or things you want changed before merging?

GeckoEidechse commented:
Technically if you already switched to release candidate channel and haven't switched back since you only need to press "Update". However, not sure how to best phrase this so honestly, just merge and I'll fix later ^^

ashleyburgers commented:
maybe have 2 lines below "installation" for if it's the first time you're installing a release candidate or if you've done it before?
